### PR TITLE
Fix the `content-type` issue for the MQTT protocol

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Fixed: https://github.com/cloudevents/sdk-go/issues/1062

### CODE REVIEW

**Comment by yanmxa** [comment]
This is a bug of the MQTT protocol in the structured mode. Please take a look @embano1 ~

File: protocol/mqtt_paho/v2/message.go [hunk 1]
```diff
@@ -42,7 +41,7 @@ func NewMessage(msg *paho.Publish) *Message {
 	var v spec.Version
 	if msg.Properties != nil {
 		// Use properties.User["Content-type"] to determine if message is structured
```
Comment (embano1) [inline protocol/mqtt_paho/v2/message.go:43]: should this comment be removed then?

File: protocol/mqtt_paho/v2/message.go [hunk 2]
```diff
@@ -88,14 +87,20 @@ func (m *Message) ReadBinary(ctx context.Context, encoder binding.BinaryWriter)
...
 			}
-		} else if userProperty.Key == contentType {
-			err = encoder.SetAttribute(m.version.AttributeFromKind(spec.DataContentType), string(userProperty.Value))
 		}
 		if err != nil {
 			return
 		}
 	}
 
+	contentType := m.internal.Properties.ContentType
```
Comment (embano1) [inline protocol/mqtt_paho/v2/message.go:96]: question: no changes needed for structured encoding?

**Review by embano1 — COMMENTED** [review]
Just a nit and a question

File: protocol/mqtt_paho/v2/message.go (hunk 2 above)
Comment (yanmxa) [reply protocol/mqtt_paho/v2/message.go]: Yes! The binary message doesn't need the content type attribute.  Remove it now.

File: protocol/mqtt_paho/v2/message.go (hunk 1 above)
Comment (yanmxa) [reply protocol/mqtt_paho/v2/message.go]: Done.

File: protocol/mqtt_paho/v2/message.go (hunk 2 above)
Comment (yanmxa) [reply protocol/mqtt_paho/v2/message.go:95]: We still need it here, cause the attribute `datacontenttype` of the event is decoded from this field with a binary message

**Review by embano1 — APPROVED** [review]
LGTM (merging once tests pass)

**Pull request merged**